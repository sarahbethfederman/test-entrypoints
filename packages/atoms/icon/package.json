{
  "name": "@auscred/icon",
  "version": "0.0.1",
  "source": "src/Icon.tsx",
  "main": "dist/Icon.js",
  "types": "dist/Icon.d.ts",
  "directories": {
    "lib": "lib"
  },
  "keywords": [],
  "author": "Lendi Devs",
  "license": "ISC",
  "devDependencies": {
    "jest": "^23.1.0",
    "@lui/story-utils": "^1.0.0",
    "@storybook/addon-knobs": "^3.4.8",
    "@types/storybook__addon-knobs": "^3.3.1"
  },
  "scripts": {
    "build": "node ./src/build.js && tsc -p ./tsconfig.json"
  },
  "dependencies": {
    "svg-sprite": "^1.4.0"
  }
}
