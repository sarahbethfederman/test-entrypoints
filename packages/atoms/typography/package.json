{
  "name": "@auscred/typography",
  "version": "0.0.1",
  "main": "dist/Typography.js",
  "types": "dist/Typography.d.ts",
  "files": [
    "dist"
  ],
  "keywords": [],
  "author": "Lendi Devs",
  "license": "ISC",
  "peerDependencies": {
    "@types/react": "^16.4.0",
    "react": "^16.4.0",
    "styled-components": "^3.3.3"
  },
  "scripts": {
    "build": "tsc --pretty"
  }
}
