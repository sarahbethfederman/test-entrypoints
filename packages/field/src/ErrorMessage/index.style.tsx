import styled from 'styled-components';
import { pt } from '@lendi-ui/spacing';

export const Wrapper = styled.div`
  width: 100%;
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: base-line;
  ${pt('xxxs')};
`;
