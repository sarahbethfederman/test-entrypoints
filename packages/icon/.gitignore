src/icons-compiled/*
src/index.ts