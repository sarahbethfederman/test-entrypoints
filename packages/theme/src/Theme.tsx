import * as React from 'react';
import styled, { ThemeProvider, injectGlobal } from 'styled-components';
import { theme } from './theme/lendi';
import { ThemeMap } from './types';

interface ThemeProps {
  kind?: ThemeMap;
  children?: React.ReactNode;
}

// Any cross browser resets should occur here
// tslint:disable-next-line no-unused-expression
injectGlobal`
  html, body {
    margin: 0;
    padding: 0;
  }

  :root {
    --lendi-ui-size: 14px;
  }

  @media (min-width: 36.0625em) {
    :root {
      --lendi-ui-size: 16px;
    }
  }
`;

const Root = styled.div`
  color: ${theme.colors.shade[700]};
`;

export function Theme(props: ThemeProps) {
  const { children } = props;
  const kind = props.kind || theme;

  return (
    <ThemeProvider theme={kind}>
      <Root>{children}</Root>
    </ThemeProvider>
  );
}
