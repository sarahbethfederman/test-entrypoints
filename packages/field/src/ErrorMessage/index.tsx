import * as React from 'react';
import { Wrapper } from './index.style';
import { Error } from '@lendi-ui/icon';
import { Body } from '@lendi-ui/typography';

export interface ErrorMessageProps {
  error?: string;
}

const ErrorMessage = ({ error }: ErrorMessageProps) => {
  const ErrorMessageColor = 'error.500';
  return (
    <Wrapper color={ErrorMessageColor}>
      <Error color={ErrorMessageColor} />
      <Body size="xs" color={ErrorMessageColor} ml="xxxs">
        {error}
      </Body>
    </Wrapper>
  );
};

export default ErrorMessage;
