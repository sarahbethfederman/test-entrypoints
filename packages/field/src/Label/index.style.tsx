import styled from 'styled-components';
import { Body, Heading } from '@lendi-ui/typography';
import { map, BreakpointValue, BreakpointValueMap } from '@lendi-ui/breakpoint';
import { deriveSize } from '@lendi-ui/utils';

export type Size = 'lg' | 'sm';
export type LabelSize = BreakpointValue<Size> | BreakpointValueMap<Size>;

const bodyStyleBySizeMixin = (size: LabelSize) =>
  map(size, (val) => {
    switch (val) {
      case 'sm':
        return `
          font-size: ${deriveSize(0.75)};
      `;
      case 'lg':
        return `
          font-size: ${deriveSize(1)};
      `;
      default:
        return undefined;
    }
  });

const headingStyleBySizeMixin = (size: LabelSize) =>
  map(size, (val) => {
    switch (val) {
      case 'sm':
        return `
          font-size: ${deriveSize(1)};
          margin-right: ${deriveSize(0.5)};
      `;
      case 'lg':
        return `
          font-size: ${deriveSize(1.75)};
          margin-right: ${deriveSize(1)};
      `;
      default:
        return undefined;
    }
  });

export const LabelHeading = styled(Heading)`
  ${({ size }: { size: LabelSize }) => headingStyleBySizeMixin(size)} display: inline-block;
`;

export const AssistiveBody = styled(Body)`
  ${({ size }: { size: LabelSize }) => bodyStyleBySizeMixin(size)};
`;
