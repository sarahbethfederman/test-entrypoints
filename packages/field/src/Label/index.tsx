import * as React from 'react';
import { LinkProps } from '@lendi-ui/typography';
import { LabelSize, LabelBody, LabelHeading, AssistiveBody } from './index.style';

export interface LabelProps {
  size?: LabelSize;
  label?: string;
  assistiveText?: string;
  link?: React.ReactElement<LinkProps>;
  isOptional?: boolean;
}

const Label = ({ size = 'sm', label, assistiveText, link, isOptional }: LabelProps) => {
  let labelBody, assistiveBody;
  switch (size) {
    case 'sm':
      labelBody = (
        <LabelBody size="md" ml="xxs">
          <strong>
            {label}
            {isOptional && ' (Optional)'}
          </strong>
        </LabelBody>
      );
      assistiveBody = assistiveText ? (
        <AssistiveBody color="shade.400" size="xs">
          {assistiveText}
        </AssistiveBody>
      ) : null;
      break;
    case 'lg':
      labelBody = (
        <LabelHeading size="md">
          {label}
          {isOptional && ' (Optional)'}
        </LabelHeading>
      );
      assistiveBody = assistiveText ? (
        <AssistiveBody color="shade.400" size="md">
          {assistiveText}
        </AssistiveBody>
      ) : null;
      break;
  }
  return (
    <div>
      {labelBody}
      {link}
      {assistiveBody}
    </div>
  );
};

export default Label;
