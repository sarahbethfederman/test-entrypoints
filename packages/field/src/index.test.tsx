import * as React from 'react';
import { mount } from 'enzyme';
import Field from './index';
import Label from './Label/index';
import ErrorMessage from './ErrorMessage/index';
import { FieldWrapper, LabelField } from './index.style';
import { QuestionOutline } from '@lendi-ui/icon'; // replace with ToolTip later when LUI Input added to LUI
import { Link } from '@lendi-ui/typography';
import Theme from '@lendi-ui/theme';

let wrapper;

function render(props) {
  wrapper = mount(
    <Theme>
      <Field {...props}>{props.children}</Field>
    </Theme>
  );
}
jest.mock('cuid', () => () => 123456789);

describe('Field', () => {
  const label = 'First name';
  const size = 'sm';
  const link = <Link size="sm">forgot?</Link>;
  const assistiveText = 'Assistive text';
  const error = 'error message';
  const tooltip = <QuestionOutline color="primary.500" />;
  const children = <input type="text" />;
  const isRequired = false;
  it('should render wrapper component', () => {
    render({ label });
    expect(wrapper.find(ErrorMessage).length).toEqual(0);
    expect(wrapper.find(FieldWrapper).length).toEqual(1);
    expect(wrapper.find(FieldWrapper)).toMatchSnapshot();
    expect(wrapper.find(LabelField).length).toEqual(1);
  });

  it('should render Label component properly', () => {
    render({ label, size, link, assistiveText });
    expect(wrapper.find(Label).length).toEqual(1);
    expect(wrapper.find(Label).prop('label')).toEqual('First name');
    expect(wrapper.find(Label).prop('size')).toEqual('sm');
    expect(wrapper.find(Label).prop('link')).toEqual(<Link size="sm">forgot?</Link>);
    expect(wrapper.find(Label).prop('assistiveText')).toEqual('Assistive text');
  });

  it('should render the error message when error is defined and the user has interacted with the component', () => {
    render({ error, touched: true });
    expect(wrapper.find(ErrorMessage)).toHaveLength(1);
    expect(wrapper.find(ErrorMessage).prop('error')).toEqual('error message');
  });

  it('should not render the error message when error is undefiend and the user has interacted with the component', () => {
    render({ touched: true });
    expect(wrapper.find(ErrorMessage)).toHaveLength(0);
  });

  it('should not render the error message when error is defined and the user has not interacted with the component', () => {
    render({ error });
    expect(wrapper.find(ErrorMessage)).toHaveLength(0);
  });

  it('should render ToolTip component properly', () => {
    render({ tooltip });
    expect(wrapper.find(QuestionOutline).length).toEqual(1); //  replace with ToolTip later when LUI Input added to LUI
  });

  it('should render children component properly', () => {
    render({ children });
    expect(wrapper.find('input').length).toEqual(1); //  replace with Input later when LUI Input added to LUI
  });

  it('should render Label component with Optional properly', () => {
    render({ label, isRequired });
    expect(wrapper.find('Label').prop('isRequired')).toEqual(false);
  });

  it('should not render Label component if no related props are passed', () => {
    render({});
    expect(wrapper.find(LabelField).length).toEqual(0);
  });

  it('should render Label component if at least one related prop is passed', () => {
    render({ label });
    expect(wrapper.find(LabelField).length).toEqual(1);
    render({ isRequired });
    expect(wrapper.find(LabelField).length).toEqual(1);
    render({ assistiveText });
    expect(wrapper.find(LabelField).length).toEqual(1);
    render({ link });
    expect(wrapper.find(LabelField).length).toEqual(1);
    render({ tooltip });
    expect(wrapper.find(LabelField).length).toEqual(1);
  });

  describe('test native props and Standard HTML Attributes', () => {
    it('should mount with Aria attributes', () => {
      const ARIA_LABEL = 'testLabel';
      const ARIA_DESCRIBE_BY = 'info';
      render({ 'aria-label': ARIA_LABEL, 'aria-describedby': ARIA_DESCRIBE_BY });
      const fieldWrapperAttributes = wrapper.find(FieldWrapper).props();
      expect(fieldWrapperAttributes['aria-label']).toBe(ARIA_LABEL);
      expect(fieldWrapperAttributes['aria-describedby']).toBe(ARIA_DESCRIBE_BY);
    });
    it('should mount with native props like id, tabIndex', () => {
      const TEXT_ID = 'testId';
      render({
        id: TEXT_ID,
        tabIndex: 1,
      });
      const fieldWrapperAttributes = wrapper.find(FieldWrapper).props();
      expect(fieldWrapperAttributes.id).toBe(TEXT_ID);
      expect(fieldWrapperAttributes.tabIndex).toBe(1);
    });
  });
});

describe('field should be a11y(accessibility) by having htmlFor', () => {
  describe('when htmlFor is provided', () => {
    const id = 'test';
    beforeEach(() => {
      render({ label: 'testLabel', htmlFor: id });
    });
    it('should render with <label> with htmlFor', () => {
      expect(wrapper.find(LabelField).props().htmlFor).toBe(id);
    });
  });
  describe('when htmlFor is NOT provided, then auto generate it and make it available at children', () => {
    beforeEach(() => {
      render({ label: 'testLabel' });
    });
    it('should render with <label> with htmlFor', () => {
      expect(wrapper.find(LabelField).props().htmlFor).toBe(123456789);
    });
  });
});
