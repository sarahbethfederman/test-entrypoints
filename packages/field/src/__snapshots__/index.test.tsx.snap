// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field it should render wrapper component 1`] = `
.c5 {
  margin-top: 0;
  margin-bottom: 0;
  color: #929292;
  font-size: calc(0.875 * var(--lendi-ui-size));
  line-height: calc(20 / 14);
  font-family: 'Open Sans',sans-serif;
}

.c3 {
  margin-top: 0;
  margin-bottom: 0;
  font-size: calc(1.375 * var(--lendi-ui-size));
  font-weight: 500;
  line-height: calc(28 / 22);
  font-family: 'Cabin',sans-serif;
}

.c2 {
  font-size: var(--lendi-ui-size);
  margin-right: calc(0.5 * var(--lendi-ui-size));
  display: inline-block;
}

.c4 {
  font-size: calc(0.75 * var(--lendi-ui-size));
}

.c0 {
  box-sizing: border-box;
  line-height: 1.333;
  white-space: normal;
  border-spacing: 2px;
  text-align: left;
  word-spacing: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  height: 100%;
}

.c1 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-bottom: calc(0.25 * var(--lendi-ui-size));
}

<styled.div>
  <div
    className="c0"
  >
    <styled.label
      htmlFor={123456789}
    >
      <label
        className="c1"
        htmlFor={123456789}
      >
        <Label
          isOptional={false}
          label="First name"
          size="sm"
        >
          <div>
            <Styled(Styled(HeadingTag))
              size="sm"
            >
              <Styled(HeadingTag)
                className="c2"
                size="sm"
              >
                <HeadingTag
                  className="c2 c3"
                  size="sm"
                >
                  <h4
                    className="c2 c3"
                    size="sm"
                  >
                    First name
                  </h4>
                </HeadingTag>
              </Styled(HeadingTag)>
            </Styled(Styled(HeadingTag))>
            <Styled(styled.p)
              color="shade.400"
              size="sm"
            >
              <styled.p
                className="c4"
                color="shade.400"
                size="sm"
              >
                <p
                  className="c4 c5"
                  color="shade.400"
                  size="sm"
                />
              </styled.p>
            </Styled(styled.p)>
          </div>
        </Label>
      </label>
    </styled.label>
  </div>
</styled.div>
`;
